controllers/util/solr_util.go

```diff
@@ -594,6 +594,21 @@ func generateSolrSetupInitContainers(solrCloud *solr.SolrCloud, solrCloudStatus
 	}
 	setupCommands := []string{"cp /tmp/solr.xml /tmp-config/solr.xml"}
 
+	// Figure out the solrUser and solrGroup to use
+	solrUser := DefaultSolrUser
+	solrGroup := DefaultSolrGroup
+	solrPodSecurityContext := solrCloud.Spec.CustomSolrKubeOptions.PodOptions.PodSecurityContext
+
+	if solrPodSecurityContext.RunAsUser != nil {
+		solrUser = int(*solrPodSecurityContext.RunAsUser)
+
+		if solrPodSecurityContext.RunAsGroup != nil {
```

Comment:
The user might specify a `runAsGroup` without a `runAsUser`, so we probably want to keep these two checks separate right? Or am I thinking about this incorrectly?

Reply:
So this should be `fsGroup` not `runAsGroup`, since `fsGroup` is meant to be used for volumes.